### ISSUE TITLE
CalorimetryCalculator class added, with tests

### ISSUE THREAD

**Issue by paul-shannon** [ryan-hoffman/calorimetry-project#2]
@ryan-hoffman   

I created a new directory (new, in order to keep it out of the way, and separate from your main code) called ````explorations````

The development style illustrated there will surely fill like overkill.  But that may a virtue:  there is little algorithmic or data structure complexity, just some file- and directory-structuring practices, plus the introduction of unit tests:
````
explorations/
   calorimeterCalculatory.py   # a simple class, ignorant of flask, dash, etc
   tests/ 
       test_calorimeteryCalculator.py  # tests several features of the class
       makefile  #  type 'make' in this directory, all the tests will run
````
These introduce the basics of engineering a software project:
  -  separation of concerns
  -  every discernible "atomic" part goes in its own file (and often, its own class) 
  -  each part is ignorant of most others, is tested separately
  -  use tests from the start, add to them as bugs appear and as features are added
Most people resist "test-driven development"  on first encounter,  but once you join the church you cannot imagine working any other way.  You write code faster, it is  much more robust, it is ready to be evolved to new forms.

**Comment by ryan-hoffman** [comment]
@paul-shannon 

Thanks for the comments and the new files.  I pulled them and will go over them, and probably come back in a few days with a bunch of questions.  But first, to run the tests... I don't understand this comment:  #  type 'make' in this directory, all the tests will run.  You mean in the command line?  I tried <make> <makefile> <python make> <python makefile> after cd'ing to /tests but it didn't run any tests, so obviously I'm doing something wrong.

**Comment by paul-shannon** [comment]
@ryan-hoffman 
````make```` is an old-school, command line indispensable programming tool.  It looks (by default) for a file called ````makefile```` in the current directory, and applies simple rules specified there to build and run software.

Here's what things look like for me:
````
shell> cd  calorimetry-project/explorations/tests
shell> make
python test_calorimetryCalculator.py
--- test_constructor
--- test_constructorWithBadData
--- test_calculation
````
I use ````shell> ```` above to indicate that you are in your OS shell  (typically bash) - but this presumes a linux-like, mabye macos operating system.   Maybe  you are on a Windows machine?  I should have inquired about that earlier.

**Comment by ryan-hoffman** [comment]
No, I'm using a Mac.  Typically a bash shell.  It starts out as an iterm window, changes to bash when I start opening or creating directories, but recently (as in with the Dash project, it throws a datatype error and the window switches to python3.  I understand the error (and it doesn't cause my app to crash) but I haven't looked into what exactly is going on with these name switches.  No need to answer that, I'll research it myself and get back to you if I have questions.  Ok, I'll mess around with the make command later (probably tonight) and see if I can get it to run the tests.   Thanks!

**Comment by paul-shannon** [comment]
(real programers use emacs!)

**Comment by ryan-hoffman** [comment]
The tests to run just fine, as far as I can tell.  The output in my shell looks just like yours.  Thanks for the instruction!
